File: interviewBit-golang/src/arrays/problems/diagonalFlip.go
package arrayProblems

/*
	Problem : https://www.interviewbit.com/problems/diagonal-flip/
*/

// T(n) : O(n) , S(n) : O(1)
func DiagonalFlip(A [][]int) [][]int {

	n := len(A)
	for i := 0; i < n; i++ {
		for j := 0; j < i; j++ {
			A[i][j], A[j][i] = A[j][i], A[i][j]
		}
	}

	return A
}

File: interviewBit-cpp/src/arrays/diagonal-flip.cpp
#include <iostream>
#include <vector>

using namespace std;

// https://www.interviewbit.com/problems/diagonal-flip/

// T(n) : O(n) ; S(n) : O(1)
vector<vector<int> > diagonalFlip(vector<vector<int> > &A) {

    int n = A.size();
    for (int i = 0; i < n; i++) {
        for (int j = i+1; j < n; j++) {
            swap(A[i][j], A[j][i]);
        }
    }

    return A;
}

// Driver Code for testing
int main() {

    return 0;
}
